src/engine/SCons/Tool/wix.py

```diff
@@ -62,39 +64,58 @@ def generate(env):
 
     env['BUILDERS']['WiX'] = linker_builder
 
+
 def exists(env):
     env['WIXCANDLE'] = 'candle.exe'
     env['WIXLIGHT']  = 'light.exe'
 
-    # try to find the candle.exe and light.exe tools and 
-    # add the install directory to light libpath.
+    def check_wix(e, p):
+        """try to find the candle and light tools; add to path if found."""
+        try:
+            files = os.listdir(p)
+            if e['WIXCANDLE'] in files and e['WIXLIGHT'] in files:
+                e.PrependENVPath('PATH', p)
+                # include appropriate flags if running WiX 2.0
+                if 'wixui.wixlib' in files and 'WixUI_en-us.wxl' in files:
+                    e['WIXLIGHTFLAGS'] = [os.path.join(p, 'wixui.wixlib'),
+                                          '-loc',
+                                          os.path.join(p, 'WixUI_en-us.wxl')]
+                else:
+                    e['WIXLIGHTFLAGS'] = []
+                return 1
+        except OSError:
+            pass  # ignore this, could be a stale PATH entry.
+        return None
+
+    # look in existing paths first
     for path in os.environ['PATH'].split(os.pathsep):
         if not path:
             continue
 
         # workaround for some weird python win32 bug.
-        if path[0] == '"' and path[-1:]=='"':
+        if path[0] == '"' and path[-1:] == '"':
             path = path[1:-1]
 
         # normalize the path
         path = os.path.normpath(path)
 
-        # search for the tools in the PATH environment variable
-        try:
-            files = os.listdir(path)
-            if env['WIXCANDLE'] in files and env['WIXLIGHT'] in files:
-                env.PrependENVPath('PATH', path)
-                # include appropriate flags if running WiX 2.0
-                if 'wixui.wixlib' in files and 'WixUI_en-us.wxl' in files:
-                    env['WIXLIGHTFLAGS'] = [ os.path.join( path, 'wixui.wixlib' ),
-                                             '-loc',
-                                             os.path.join( path, 'WixUI_en-us.wxl' ) ]
-                else:
-                    env['WIXLIGHTFLAGS'] = []
-                return 1
-        except OSError:
-            pass # ignore this, could be a stale PATH entry.
-
+        if check_wix(env, path):
+            return 1
+
+    # Wix is not in scons' PATH. Let's look in typical install locations.
+    wix_env = os.environ.get('WIX', None)
```

Comment:
I think the construction environment should tend not to be dependent on the system environment?